app/policies/registration_policy.rb

```diff
@@ -0,0 +1,24 @@
+class RegistrationPolicy
+  def initialize(user, params)
+    @user = user
+    @params = params
+  end
+
+  def create?
+    email_registrable?
+  end
+
+  private
+
+  def email_registrable?
+    if Settings::General.waiting_on_first_user
+      if ENV["FOREM_OWNER_SECRET"].present?
+        ENV["FOREM_OWNER_SECRET"] == @params[:user][:forem_owner_secret]
```

Comment:
I'm sure `@params[:user]` is expected to be a hash if we're checking a policy - in the case where it's not a hash this will raise an error. Would dig be needlessly defensive?

```
@params.dig(:user, :forem_owner_secret)
```

Reply:
This is a good idea! updating.
```suggestion
        ENV["FOREM_OWNER_SECRET"] == @params.dig(:user, :forem_owner_secret)
```